### PR TITLE
feature(authentication) add refresh token

### PR DESCRIPTION
**Pull request by wfajczyk**
# Description

Issue #882
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

- [ ] Unit Tests
- [x] Behat Test

# Checklist:
- [x] I have read the contribution requirements and fulfill them.
- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: module/core/src/Test/Behat/Context/ApiAuthTokenContext.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    {
+        $this->request = $request;
+    }
+
+    /**
+     * @Given I use Authenticated token :token
+     */
+    public function iAmAuthenticatedAsUser(string $token): void
+    {
+        $this->request->setHttpHeader('JWTAuthorization', 'Bearer '.$token);
```
Comment (piotrkreft) [inline module/core/src/Test/Behat/Context/ApiAuthTokenContext.php:28]: the token name depends on the configuration, should not be hardcoded here

File: module/authentication/src/Application/Security/Provider/RefreshUserProvider.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+namespace Ergonode\Authentication\Application\Security\Provider;
+
+use Ergonode\Account\Domain\Repository\UserRepositoryInterface;
+use Ergonode\Authentication\Application\Security\User\User;
+use Ergonode\SharedKernel\Domain\Aggregate\UserId;
+use Symfony\Component\Security\Core\Exception\UsernameNotFoundException;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+
+class RefreshUserProvider implements UserProviderInterface
```
Comment (piotrkreft) [inline module/authentication/src/Application/Security/Provider/RefreshUserProvider.php:18]: Based on other providers I guess IdUserProvider would be more consistent - the provider does not need to know the context it's being used in.

**Review by piotrkreft — CHANGES_REQUESTED** [review]

File: module/authentication/src/Application/Security/Provider/RefreshUserProvider.php (hunk 2 above)
Comment (wfajczyk) [reply module/authentication/src/Application/Security/Provider/RefreshUserProvider.php]: Ok, Thanks for the tip.

File: module/core/src/Test/Behat/Context/ApiAuthTokenContext.php (hunk 1 above)
Comment (wfajczyk) [reply module/core/src/Test/Behat/Context/ApiAuthTokenContext.php]: I changed to the `behat` configuration

File: behat.yml.dist [hunk 3]
```diff
@@ -45,6 +45,9 @@ default:
                 - Ergonode\Core\Test\Behat\Context\ApiAuthContext:
                     JWTTokenManager: '@lexik_jwt_authentication.jwt_manager'
                     request: '@Behatch\HttpCall\Request'
+                - Ergonode\Core\Test\Behat\Context\ApiAuthTokenContext:
+                    request: '@Behatch\HttpCall\Request'
+                    authHeader: 'JWTAuthorization'
```
Comment (piotrkreft) [inline behat.yml.dist:50]: Perhaps set it as the container parameter and the value can be easily shared and set in just one place?

**Review by piotrkreft — APPROVED** [review]

**Review by rprzedzik — APPROVED** [review]

**Pull request merged**